aiplatform/src/main/java/aiplatform/PredictCodeChatSample.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aiplatform;
+
+// [START aiplatform_sdk_code_chat]
+
+import com.google.cloud.aiplatform.v1beta1.EndpointName;
+import com.google.cloud.aiplatform.v1beta1.PredictResponse;
+import com.google.cloud.aiplatform.v1beta1.PredictionServiceClient;
+import com.google.cloud.aiplatform.v1beta1.PredictionServiceSettings;
+import com.google.protobuf.Value;
+import com.google.protobuf.util.JsonFormat;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+// Use a code chat model to generate a function
+public class PredictCodeChatSample {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String instance =
+        "{ \"messages\": [\n"
+            + "{\n"
+            + "  \"author\": \"content\",\n"
+            + "  \"content\": \"Hi, how are you?\"\n"
+            + "},\n"
+            + "{\n"
+            + "  \"author\": \"system\",\n"
+            + "  \"content\": \"I am doing good. What can I help you in the coding world?\",\n"
+            + " },\n"
+            + "{\n"
+            + "  \"author\": \"content\",\n"
+            + "  \"content\":\n"
+            + "     \"Please help write a function to calculate the min of two numbers.\",\n"
+            + "},\n"
+            + "],";
+    String parameters = "{\n" + "  \"temperature\": 0.2,\n" + "  \"maxOutputTokens\": 256,\n" + "}";
+    String project = "YOUR_PROJECT_ID";
+    String location = "us-central1";
+    String publisher = "google";
+    String model = "codechat-bison@001";
+
+    predictCodeChat(instance, parameters, project, location, publisher, model);
+  }
+
+  static void predictCodeChat(
+      String instance,
+      String parameters,
+      String project,
+      String location,
+      String publisher,
+      String model)
+      throws IOException {
+    String endpoint = String.format("%s-aiplatform.googleapis.com:443", location);
```

Comment:
Hard-coding endpoints is an anti-pattern. Is it possible to avoid explicit definition of the endpoint? Please, consider using discovery service or other means of obtaining the endpoint instead of hard-coding it.

Reply:
As discussed, there is no good solution at the moment. We will consider this as future improvement.